.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,174 @@
+import { ApiItem, ApiTypeResult, ExtensionApiMap } from '../types';
+import * as babel from '@babel/core';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { singularize } from '../utils/string';
+import { loadExtensionApis } from './api-loader';
+import os from 'os';
+import { ParallelController } from '../utils/parallel';
+
+const CPU_CORES = os.cpus().length;
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+export const getApiType = (
+  apiCategory: string,
+  apiName: string
+): ApiTypeResult => {
+  apiCategory = apiCategory.replace(/_/g, '.');
+
+  if (EXTENSION_API_MAP[apiCategory]) {
+    const apiTypes = EXTENSION_API_MAP[apiCategory];
+
+    for (let type of Object.keys(apiTypes)) {
```

Comment:
I actually think individually checking things here might be easier to read. e.g:

```ts
if (apiTypes.functions.includes(apiName) {
  return "method";
}

if (apiTypes.properties.includes(apiName) {
  return "property";
}

...
```